SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -61,20 +61,46 @@ function Get-Secret
         $Name = $Matches[1]
     }
 
-    $res = Invoke-lpass 'show','--name', $Name, '--password'
-    if ([string]::IsNullOrWhiteSpace($res)) {
-        $res = Invoke-lpass 'show', '--name', $Name, '--notes'
-    } else {
-        # We have a password, check for a username
-        $username = Invoke-lpass 'show', '--name', $Name, '--username'
-        if ($username) {
-            $res = [System.Management.Automation.PSCredential]::new(
-                $username,
-                (ConvertTo-SecureString $res -AsPlainText -Force))
-        }
+    try {
+        $res = Invoke-lpass 'show', '--name', $Name, '--all'
     }
+    catch {
+        Write-Error $_
+        return 
+    }
+    
+    $Raw = ($res | Select-Object -Skip 1) -join "`n"
+
+    # Custom type cannot have the same case-sensitive name but Username / uSERname work.
+    # That's why we don't parse it directly to a hashtable.
```

Comment:
Maybe use this solution for a case-sensitive hashtable?

https://stackoverflow.com/a/24054257

If that doesn't work, maybe just use a Dictionary:

```pwsh
$myDict = [System.Collections.Generic.Dictionary[string,string]]::new()
$myDict['foo'] = "bar"
$myDict['FOo'] = "bar"
```

Reply:
Ohhh indeed. I'll test that.